packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -239,6 +250,10 @@ export class Distribution extends Resource implements IDistribution {
       if (!Token.isUnresolved(certificateRegion) && certificateRegion !== 'us-east-1') {
         throw new Error(`Distribution certificates must be in the us-east-1 region and the certificate you provided is in ${certificateRegion}.`);
       }
+
+      if (!props.domainNames || props.domainNames.length === 0) {
```

Comment:
I usually write this as:

```suggestion
      if ((props.domainNames ?? []).length === 0) {
```

I like it a little bit more, because it doesn't repeat the main expression (`prop.domainNames` in this case).